pkg/segment/metadata/metadata_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package metadata
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"testing"
+
+	localstorage "github.com/siglens/siglens/pkg/blob/local"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_AddSementInfo(t *testing.T) {
+	_ = localstorage.InitLocalStorage()
+	ResetGlobalMetadataForTest()
+	for i := uint64(0); i < 10; i++ {
+		currInfo := &SegmentMicroIndex{
+			SegMeta: structs.SegMeta{
+				LatestEpochMS:    i,
+				VirtualTableName: "test-1",
+				SegmentKey:       strconv.FormatUint(i, 10),
+			},
+		}
+		BulkAddSegmentMicroIndex([]*SegmentMicroIndex{currInfo})
+	}
+
+	for i := int64(0); i < 10; i++ {
+		assert.Equal(t, uint64(i), GetSegmentMetadataReverseIndexForTest()[strconv.FormatInt(i, 10)].LatestEpochMS)
+	}
+	nextVal := uint64(9)
+	for i := 0; i < 10; i++ {
+		assert.Equal(t, GetAllSegmentMicroIndexForTest()[i].LatestEpochMS, nextVal)
+		log.Infof("i %+v latest: %+v", i, GetAllSegmentMicroIndexForTest()[i].LatestEpochMS)
+		nextVal--
+	}
```

Comment:
I'm missing something. For the second loop, why is the expected value `9, 8, 7, ...` instead of `0, 1, 2, ...`?

When you made the mirco indexes, you set the `i`-th one to have a `LatestEpoch` of `i`, and that's what your first `for` tests here.

Reply:
The second for loop tests the functionality of this func `GetAllSegmentMicroIndexForTest` to make sure it returns the microindices according the latestEpoch sorted by time (most recent first), since that is what is used during query time. We want to look at the segment that is most recent